addon/components/search-facet-worktype/component.js

```diff
@@ -0,0 +1,61 @@
+import Ember from 'ember';
+import layout from './template';
+import { termsFilter, getUniqueList } from '../../utils/elastic-query';
+
+/**
+ * Copied from Ember-SHARE.  Type facet.
+ *
+ * ```handlebars
+ * {{search-facet-worktype
+ *      key=facet.key
+ *      options=facet
+ *      aggregations=aggregations
+ *      state=(get facetStates facet.key)
+ *      filter=(get facetFilters facet.key)
+ *      onChange=(action 'facetChanged')
+ * }}
+ * ```
+ * @class search-facet-worktype
+ */
+export default Ember.Component.extend({
+    layout,
+    init() {
+        this._super(...arguments);
+        this.send('setState', this.get('state'));
+    },
+
+    selected: Ember.computed('state', function() {
+        return this.get('state') || [];
+    }),
+
+    changed: Ember.observer('state', function() {
+        let state = Ember.isBlank(this.get('state')) ? [] : this.get('state');
+        let previousState = this.get('previousState') || [];
+
+        if (Ember.compare(previousState, state) !== 0) {
```

Comment:
I just copied this from Ember-SHARE, but I am guessing it was used to be able to compare arrays.  Even if both arrays are equal, comparing previousState !== state will be true.